docs/en/contributor/Code-Conventions.md

```diff
@@ -123,39 +94,436 @@ public MyClass {
+* Any degree of detail within an experimental or new feature
+
+**During Exception Handling**
+* Up to the developer’s discretion
+
+## Exceptions
+
+### Throwing Exceptions
+
+#### Use unchecked exceptions for events which indicate bugs
+
+Examples: Illegal states, invalid API usage
+
+```java
+public void method(String neverSupposedToBeNull) {
+  Preconditions.checkNotNull(neverSupposedToBeNull, "neverSupposedToBeNull");
+}
+```
+
+#### Use checked exceptions for everything else
+
+Examples: File not found, interrupted exception, timeout exceeded
+
+```java
+// Recommended: the checked exception should just be propagated
+public void handleRawUserInput(String date) throws InvalidDateException {
+  parseDate(date);
+}
+
+// Don't do this - it's reasonable for user input to be invalid.
+public void handleRawUserInput(String date) throws InvalidDateException {
+  try {
+    parseDate(date);
+  } catch (InvalidDateException e) {
+    throw new RuntimeExcepiton("date " + date + " is invalid", e);
+  }
+}
+```  
+
+#### Design code to minimize use of checked exceptions
+
+Require callers to validate inputs so that invalid arguments can be unchecked exceptions.
+
+#### Reuse existing exceptions when appropriate
+
+Try to find an appropriate existing exception before inventing a new one.
+
+#### Selectively javadoc exceptions
+Only write exception javadoc when you think it will be useful to the developer using the method.
+There are so many sources of IOException that it's almost never useful to include javadoc for it.
+
+#### AlluxioStatusExceptions
+
+On the wire we represent exceptions with one of 14 status codes, e.g. NOT_FOUND, UNAVAILABLE. Within
+our server and client code, we represent these exceptions using exception classes corresponding to
+these statuses, e.g. NotFoundException and UnavailableException. AlluxioStatusException is the
+superclass for these Java exceptions.
+
+### Handling exceptions
+
+#### Never swallow an unchecked exception.
+
+Either log it, propagate it, or log it and kill the system
+
+It is usually wrong to both log and propagate. If every method did this, the same exception would
+be logged dozens of times, polluting the logs. The responsibility for logging lies with whatever
+method ends up handling the exception without propagating it.
+
+#### Do not use Throwables.propagate()
+
+See this
+[explanation on why Throwables.propagate is deprecated](https://github.com/google/guava/wiki/Why-we-deprecated-Throwables.propagate).
+
+#### Never Ignore InterruptedException
+
+InterruptedException means that another thread has signalled that this thread should die. There are
+a few acceptable ways to handle InterruptedException, listed in order of preference.
+
+**Actually stop the thread**
+
+```java
+class MyRunnable implements Runnable {
+  public void run() {
+    while (true) {
+      try {
+        doSomethingThatThrowsInterruptedException();
+      } catch (InterruptedException e) {
+        break;
+      }
+    }
+  }
```

Comment:
one more closing brace to close the class

Reply:
Done